test/release/examples/primers/classes.chpl

```diff
@@ -63,14 +80,54 @@ class D: C {
 // Because the class ``D`` is derived from ``C``, the variable ``foo`` can
 // reference an object of type ``D``.  If an overridden method such as
 // ``printFields`` is called, it is dynamically dispatched to the method
-// with the most specific dynamic type.  The ``delete`` keyword can be
-// used to free memory associated with ``foo`` before pointing it an instance
-// of ``D``.
+// with the most specific dynamic type.
 //
-delete foo;
-foo = new unmanaged D(3, 4);
+// Note that since ``foo`` is an ``owned C``, assigning to it
+// will delete the previous instance "owned" by that variable.
+foo = new owned D(3, 4);
 foo.printFields();
-delete foo;
+
+
+// A class type includes a memory management strategy. The currently supported
+// strategies are ``owned``, ``shared``, ``unmanaged``, and ``borrowed``.
+var unm: unmanaged C = new unmanaged C();
+// 'unm' refers to a manually managed instance. It needs to have ``delete``
+// called on it to free the memory.
+delete unm;
+
+var own: owned C = new owned C(1, 10);
+// the instance referred to by 'own' is deleted at end of scope
+
+var share: shared C = new shared C(1, 10);
+// the instance referred to by 'share' is reference counted
+
+var tmp: borrowed C = new borrowed C(1, 10);
+// the instance referred to by 'tmp' will be deleted at the end of scope
```

Comment:
Same comment about "of scope" here.  I would also be tempted to note that this is not the most common way borrows are created (but I'm fine with leaving that off since you go more into borrows immediately)